code/__HELPERS/mobs.dm

```diff
@@ -231,13 +231,19 @@ GLOBAL_LIST_EMPTY(species_list)
 		checked_health["health"] = health
 	return ..()
 
+///Checks to see if our mob is currently in a do_after with the target as the target
+/mob/proc/is_interacting_with(atom/target)
+	return(target && (target in do_afters))

```

Comment:
```suggestion
	return (target in do_afters)
```
If `is_interacting_with()` is invoked with a `target` then there's flawed logic around.